File: golang/141_linked_list_cycle.go
package main

// ListNode Definition for singly-linked list.
type ListNode struct {
	Val  int
	Next *ListNode
}

func hasCycle(head *ListNode) bool {
	if head == nil || head.Next == nil {
		return false
	}
	fastPtr, slowPtr := head, head
	for fastPtr != nil && fastPtr.Next != nil {
		fastPtr = fastPtr.Next.Next
		slowPtr = slowPtr.Next

		if fastPtr == slowPtr {
			return true
		}
	}
	return false
}

File: python/141_linked_list_cycle.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def hasCycle(self, head: ListNode) -> bool:
        if head == None or head.next == None:
            return False
        fast_ptr = head
        slow_ptr = head
        while fast_ptr != None and fast_ptr.next != None:
            fast_ptr = fast_ptr.next.next
            slow_ptr = slow_ptr.next
      
            if fast_ptr == slow_ptr:
                return True
        return False